src/Mono.Android/Xamarin.Android.Net/X509TrustManagerWithValidationCallback.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+
+using Javax.Net.Ssl;
+
+using JavaCertificateException = Java.Security.Cert.CertificateException;
+using JavaX509Certificate = Java.Security.Cert.X509Certificate;
+
+namespace Xamarin.Android.Net
+{
+	internal sealed class X509TrustManagerWithValidationCallback : Java.Lang.Object, IX509TrustManager
+	{
+		internal sealed class Helper
+		{
+			public Func<HttpRequestMessage, X509Certificate2?, X509Chain?, SslPolicyErrors, bool> Callback { get; }
```

Comment:
Should `Func<>` be perhaps replaced by a specific delegate to be used throughout this class but also in `AndroidMessageHandler`? It would make it easier to modify it in the future, should there be a need for that and also reduce the signal to noise ratio when reading the code

Reply:
If exposed publicly from a public type, you don't necessarily want "easier to modify", as changing the delegate signature is a breaking change.  This should to be the same delegate type as the public `AndroidMessageHandler.ServerCertificateCustomValidationCallback` property, and thus is a type that *cannot* be changed.